### ISSUE TITLE
Stack test hanging

### ISSUE THREAD

**Issue by subfuzion** [appcelerator-archive/amp#344]
Stack test hangs toward the end while attempting to remove a network:

```
$ go test -v github.com/appcelerator/amp/api/rpc/stack
..... lots of tests passing ....
Stopping stack a2170c0d58780278d0feaa779560291012561e670dd3184100e30fe752374e5a
Service removed 263mpbo8qv8ezql7qp87xg2wo
Service removed e28sysl9xc0dmf67fhu86sq6b
Service removed 3bi9cnxs7t7drwvnuelfpdq0y
Service removed 4fp6prxra265dq33pqlknjdz3
Removing network: b4fkbc4sir6jtwebpchvmph6v
...HANGS.....
```

At the same time, other Docker operations appear to be blocked (such as `docker network ls`). This is possibly related to [this issue](https://github.com/docker/docker/issues/27323), which has been fixed [here](https://github.com/docker/docker/pull/27395) and due to be released with Docker 1.12.3.

When this happens, the only way to get out of this state is to restart the docker engine. Sometimes, that's not enough for Docker for Mac (Version 1.12.2-beta28 (12906)) and requires resetting to initial settings (losing all cached images).

If this issue can't be resolved now, then we should at least ensure the operation context has a timeout so that the process doesn't hang indefinitely.

**Comment by freignat91** [comment]
Some observations regarding this issue:
- the stack test which shows the issue is a test on network stack 1 feature (creating private and public network and remove them during stack up and stop, (nothing to do with network stage 2 or 3)
- it happened after moving to docker 1.12.2, on docker 1.12.1 we won't get hanging so often using stacks stage 1 before we move version.
- it's not systematic, but it happened for sure after a while. we make tests run with success on Ubuntu and mac, but after a given number of this test repetition, we got docker hanging
- the hanging is Docker internal, there is no way to prevent it, removing a network several times seems enough to trigger it

Tracks, i'm going to dig on:
- verify if the docker hanging is related to some specific network parameters
- verify if the docker hanging is related to the way the network are created trying to get docker hanging with only network docker commands

**Comment by freignat91** [comment]
These updates have been done to lower the docker freeze risk:
- wait after deleting a service that the service is well deleted
- wait afer deleting a network that the network is well deleted

**Comment by freignat91** [comment]
Even, with the previous update, the freeze continue to occur too often, so this workaround have been temporary apply, until the 1.12.3 fix the docker issue:
- do not delete networks
- when a network is created if it already exist, it's reused

The freeze frequency lower a lot with this workaround, it still occur on a network list this time.

**Comment by freignat91** [comment]
Fixed by Docker 1.12.3 and PR#389
